source/hp/dof_handler.cc

```diff
@@ -1286,7 +1286,18 @@ namespace hp
     Assert (tria->n_levels() > 0,
             ExcMessage("The Triangulation you are using is empty!"));
 
-    finite_elements = &ff;
+    // Only recreate the FECollection if we don't already store an instance
+    // that constains the same FiniteElement objects.
```

Comment:
I don't actually understand the logic here -- the class *already* only stores a reference/pointer to the object owned by user code. All you do here is possibly resetting a pointer, which is so cheap that it's not worth checking.